src/pgrn-custom-scan.c

```diff
@@ -855,6 +857,8 @@ PGrnMakeTupleTableSlot(CustomScanState *customScanState,
 		}
 		ttsIndex++;
 	}
+	if (snapshot != SnapshotAny)
```

Comment:
Do we need this `if`? It seems that `snapshot` is never `SnapshotAny`.

Reply:
I think that this `if` is needless too.
Because PostgreSQL provide `NULL` check in `UnregisterSnapshot()`(ref: https://github.com/postgres/postgres/blob/REL_18_1/src/backend/utils/time/snapmgr.c#L866 ).

Also, if error occurs in `RegisterSnapshot()`, `RegisterSnapshot()` returns `InvalidSnapshot`.( `InvalidSnapshot` is `NULL`.)

ref:
https://github.com/postgres/postgres/blob/REL_18_1/src/backend/utils/time/snapmgr.c#L825
https://github.com/postgres/postgres/blob/REL_18_1/src/backend/utils/time/snapmgr.c#L840
https://github.com/postgres/postgres/blob/REL_18_1/src/include/utils/snapshot.h#L119